tests/lib/rules/no-useless-assignment.js

```diff
@@ -353,7 +369,39 @@ ruleTester.run("no-useless-assignment", rule, {
         } = obj;
         console.log(a, b);`,
         `let { a, b: {c = a} = {} } = obj;
-        console.log(c);`
+        console.log(c);`,
+        {
+            code: `/*eslint test/jsx:1*/
+                function App() {
+                    const A = "";
+                    return <A/>;
+                }
+            `,
+            languageOptions: {
+                parserOptions: {
+                    ecmaFeatures: {
+                        jsx: true
+                    }
+                }
+            }
+        },
+        {
+            code: `/*eslint test/jsx:1*/
+                function App() {
+                    let A = "";
+                    foo(A);
+                    a = "A";
```

Comment:
This test is passing without the code change. Should this line maybe be this:

```suggestion
                    A = "a";
```

Reply:
oops. sure! Thank you.  I made the change. https://github.com/eslint/eslint/pull/19200/commits/41e2e34ef84648f8ffff0c5f31e8299c15089888